player.cpp

```diff
@@ -2,12 +2,13 @@
 
 #include <iostream>
 #include <fstream>
-#include <regex>
 
 Player::Player(const std::string &name, unsigned short hp, unsigned short damage) : name{name}, hp{hp}, damage{damage}
 {
 }
 
+std::regex Player::jsonParseRegex("\\s*\"([a-z]*)\"\\s*:\\s*\"?([\\w]*)\"?\\s*[,}]\\s*");
```

Comment:
Nem így gondoltam. Maradhatott volna a függvény scope-jában, csak egy `static` kulcsszót kellett volna hozzáírni. Mivel a cpp fájlban van, a global scope sem zavar annyira, de a `const` eltűnt, az már jobban zavar.

Reply:
(elnéztem, nem global, hanem class scope, de akkor is jobb a function scope, meg akkor nem kell a felesleges include sem a headerbe)

Reply:
Visszatettem a functionbe, nem tudtam, hogy lehet ilyen scopeban használni, azért került classba.

Reply:
pedig [kellett volna tudni (11-12. dia)](http://rs1.sze.hu/~hatwagnf/prg/ea09.pdf#page=11)